plugins/msr/fu-plugin-msr.c

```diff
@@ -290,11 +290,89 @@ fu_plugin_add_security_attr_dci_locked(FuPlugin *plugin, FuSecurityAttrs *attrs)
 	fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_LOCKED);
 }
 
+static gboolean
+fu_plugin_msr_safe_kernel_for_sme(FuPlugin *plugin, GError **error)
+{
+	g_autofree gchar *min = fu_plugin_get_config_value(plugin, "MinimumSmeKernelVersion");
+
+	if (min == NULL) {
+		g_debug("Ignoring kernel safety checks");
+		return TRUE;
+	}
+	return fu_common_check_kernel_version(min, error);
+}
+
+static gboolean
+fu_plugin_msr_kernel_enabled_sme(GError **error)
+{
+	g_autofree gchar *buf = NULL;
+	gsize bufsz = 0;
+	if (!g_file_get_contents("/proc/cpuinfo", &buf, &bufsz, error))
+		return FALSE;
+	if (g_strstr_len(buf, (gssize)bufsz, "sme") == NULL) {
```

Comment:
isn't this going to match smev, smex and whatever new flags are added in the future? Also, doesn't this belong in the cpu plugin (which uses the cpuinfo flags as-reported) rather than the MSR one now?

Reply:
I still wanted to check msr first for the attribute, but will look into what it takes for them to work together.

Reply:
No, it's not in leaf 0x7.  This case is the hardware feature is detected by the MSR, and it's intentionally removed from CPU feature flags when the kernel doesn't enable it.